pkg/noderesourcetopology/pluginhelpers.go

```diff
@@ -80,30 +85,91 @@ func initNodeTopologyInformer(tcfg *apiconfig.NodeResourceTopologyMatchArgs, han
 }
 
 func createNUMANodeList(zones topologyv1alpha2.ZoneList) NUMANodeList {
-	nodes := make(NUMANodeList, 0, len(zones))
-	for _, zone := range zones {
+	// maxNUMAId is 64 just init array with this len
+	numaIDToZoneIDx := make([]int, maxNUMAId)
+	nodes := NUMANodeList{}
+	// filter non Node zones and create idToIdx lookup array
+	for i, zone := range zones {
 		if zone.Type != "Node" {
 			continue
 		}
-		var numaID int
-		_, err := fmt.Sscanf(zone.Name, "node-%d", &numaID)
+
+		numaID, err := getID(zone.Name)
 		if err != nil {
-			klog.ErrorS(nil, "Invalid zone format", "zone", zone.Name)
-			continue
-		}
-		if numaID > 63 || numaID < 0 {
-			klog.ErrorS(nil, "Invalid NUMA id range", "numaID", numaID)
+			klog.Error(err)
 			continue
 		}
+
+		numaIDToZoneIDx[numaID] = i
+
 		resources := extractResources(zone)
 		klog.V(6).InfoS("extracted NUMA resources", stringify.ResourceListToLoggable(zone.Name, resources)...)
 		nodes = append(nodes, NUMANode{NUMAID: numaID, Resources: resources})
 	}
+
+	// iterate over nodes and fill them with Costs
+	for i, node := range nodes {
+		nodes[i] = *node.WithCosts(extractCosts(zones[numaIDToZoneIDx[node.NUMAID]].Costs))
```

Comment:
nit: I wonder, but it's really really minor if this is nicer:
```
for idx := range nodes {
    node := &nodes[idx]
    node.WithCosts(extractCosts(zones[numaIDToZoneIDx[node.NUMAID]].Costs))
}
```
(written on the fly, may be buggy). The gist is to edit in place vs making a copy.
Your call, I'm ok both approaches, and performance is NOT a concern here.